### PR TITLE
Add SWIG 3.0.12 as swig@3 and remove swig@3.04

### PR DESCRIPTION
**Pull request by golddranks**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
There is another PR open, ( https://github.com/Homebrew/homebrew-core/pull/39929 ) but there were some things to fix, and since I don't know when the author's going to respond, I figured that I'd sent my own. (Limiting it to Swig changes and excluding the Subversion changes present in the other PR.)

The formula is basically the same than `swig` was before SWIG 4.0.0 was introduced, expect for the removal of `head` block and addition of `keg_only :versioned_formula`.

See the rationale for the versioned formula in the discussion of https://github.com/Homebrew/homebrew-core/pull/39929 .

### CODE REVIEW

File: Formula/swig@3.rb [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+    sha256 "37bf242aad0c18317cdaef66218483c04fa57e091b7c7f9d72089f5002881338" => :el_capitan
+    sha256 "3443dbf17f78be0cecb5419772c71bb418caa91763590072224c196a57317717" => :yosemite
   end
 
   keg_only :versioned_formula
 
   depends_on "pcre"
 
   def install
+    system "./autogen.sh" if build.head?
```
Comment (zbeekman) [inline Formula/swig@3.rb]:
```suggestion
```

This needs to go: building HEAD from versioned formula is unsupported and disallowed.

File: Formula/swig@3.rb [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
-    sha256 "fa6edd5df156b7a985bfe6f74f56cf15682af6b01d11f0ed342f2f05444234f0" => :high_sierra
-    sha256 "51b8115c0a2813caf115fb5f082e7aa1b84b7c51acec70822b8619e918969a14" => :sierra
-    sha256 "84988480f25dea3fce15b6ca0e9a3322222cc473066f23626d59b20001574fdf" => :el_capitan
-    sha256 "f5e5251fa5d8f6ce3a7e63c9f3762ccd3bbc49f4664d1d781b71425992c2425b" => :yosemite
+    sha256 "7307b4ffe3222715b2206e6477c5e3022881a730eb95a717d41a3df8e6e20455" => :mojave
+    sha256 "c0e2656fd10d57281280d20ce8bf9a060cf8714f4283dd1dfde383b3688d9ed1" => :high_sierra
+    sha256 "68cb1b6bc898f2a1bd39ae24dd0235f68ffa56d04ba8cd4424835335202977d1" => :sierra
+    sha256 "37bf242aad0c18317cdaef66218483c04fa57e091b7c7f9d72089f5002881338" => :el_capitan
+    sha256 "3443dbf17f78be0cecb5419772c71bb418caa91763590072224c196a57317717" => :yosemite
   end
```
Comment (zbeekman) [inline Formula/swig@3.rb]: Please completely remove the `bottle do` block. From brew-testbot's perspective this is a new formula.

**Review by zbeekman — CHANGES_REQUESTED** [review]
@golddranks: thanks for the PR. Can you please:

- [x] Delete the `bottle do` block
- [x] Remove the `system "./autogen.sh" if build.head?` line
- [x] Add an alias (symbolic link) to https://github.com/Homebrew/homebrew-core/tree/master/Aliases `Aliases/swig@4 -> ../Formula/swig.rb`
- [x] Rebase your commits so that:
      - Deletion of swig@3.0.4 happens first with the commit message: `swig@3.04: Remove unmaintained formula`
      - Addition of `swig@3` and `swig@4` alias happens in the second commit with message header: `swig@3 3.0.12 (new formula)` followed by a blank line and any further messages you would like to add (but you could just leave it at that).

**Comment by zbeekman** [comment]
@golddranks FYI, for the new formula you could just copy the one out of the old PR: https://github.com/Homebrew/homebrew-core/pull/39929/files#diff-2adb42ed307c4ae8955ba187eff99f60 That should address everything other than the alias and my comments about the commit messages and order above.

**Comment by golddranks** [comment]
All right, rebased. Force-pushing now.

**Review by zbeekman — COMMENTED** [review]
LGTM

**Review by zbeekman — APPROVED** [review]

**Comment by zbeekman** [comment]
FYI, we're having some CI issues, so this may not get merged for a few hours. Hopefully it won't be longer than that.

**Comment by scpeters** [comment]
@BrewTestBot test this please

**Comment by golddranks** [comment]
It still seems to fail at `pull --clean https://github.com/Homebrew/homebrew-core/pull/40882` Is this spurious or is there something wrong with the branch/some cached values? Should I re-send the PR?

**Comment by zbeekman** [comment]
> Should I re-send the PR?

No, please don't. This is the symptom of our CI infra troubles. I think we need to rebuild the worker VMs again. If no re-runs this PR within a day or two, please ping us to check in.

**Comment by fxcoudert** [comment]
Thank you @golddranks

**Pull request closed without merging**